src/utils/__tests__/getMemberExpressionValuePath-test.js

```diff
@@ -53,6 +53,48 @@ describe('getMemberExpressionValuePath', () => {
         .toBe(def.parent.get('body', 1, 'expression', 'right'));
     });
 
+    it('finds nested property definitions', () => {
+      var def = statement(`
+        module.exports.Foo = () => {};
+        module.exports.Foo.propTypes = {};
+      `);
+
+      expect(getMemberExpressionValuePath(def, 'propTypes'))
+        .toBe(def.parent.get('body', 1, 'expression', 'right'));
+    });
+
+    it('finds the correct nested property definitions', () => {
+      var def = statement(`
+        module.exports.Foo = () => {};
+        module.exports.Foo.propTypes = {};
+        exports.Foo.propTypes = {};
+      `);
+
+      expect(getMemberExpressionValuePath(def, 'propTypes'))
+        .toBe(def.parent.get('body', 1, 'expression', 'right'));
+    });
+
+    it('finds property definitions that are part of a commonJS export', () => {
+      var def = statement(`
+        function Foo () {}
+        module.exports.Foo.propTypes = {};
```

Comment:
Overall I like it, but this, which I guess is what you were referring to, worries me a bit. This almost seems like too much magic (no pun intended wrt to your user name ;) )here.

In general I think the handler should be able to detect these cases:

```js
// Assignment to variable
var foo = <component definition>;
foo.propTypes = ...;

// Assignment to property
foo.bar = <component definition>;
foo.bar.propTypes = ...;

// In both cases we should be able to detect aliasing
var bar = foo; // = foo.bar;
bar.propTypes = ...;

something.else = foo; // = foo.bar;
something.else.propTypes = ...;
```

This is going to be a bit more elaborate I think. It basically requires us to 

- remember in which scope a component was defined
- look for assignments of the variable or member expression to another variable / member expression in that scope
- look for assignments to member expressions where the object is either the original variable / member expression or one of its aliases.